### PR TITLE
Dis/enable beam centre component boxes on component update

### LINKED ISSUE #25997: Incorrect components selected in beam centre finder
The beam centre finder tab in the ISIS SANS gui allows a user to find centres using HAB and LAB components. If you have selected to reduce from e.g. HAB,  the beam centre finder still tries to use HAB and LAB by default.
If we have selected to reduce by HAB, the LAB checkbox should be unchecked and disabled by default, and vice versa.

### Steps to reproduce the behavior
1. Open up interfaces -> SANS -> ISIS SANS
2. Go to the settings tab, and select either front or rear detector (HAB and LAB)
3. Go to beam centre tab. The HAB and LAB checkboxes will both still be selected

### PR DESCRIPTION
**Pull request by TTitcombe**
**Description of work.**
This PR automatically updates the HAB/LAB component checkboxes in the ISIS SANS beam centre finder tab if the reduction mode has been changed.

If HAB mode has been selected, the beam centre finder LAB checkbox is unchecked and disabled, and vice versa. This change clarifies which components have been selected for reduction.

**Report to:** Steve King ISIS

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. In the settings tab, select the HAB reduction mode ("Hab" or "front")
3. Go to the beam centre finder tab. The HAB/Hab/front checkbox should be enabled and checked. The LAB/main-detector/rear checkbox should be unchecked and disabled
4. Back in settings, select LAB reduction mode ("main-detector" or "rear")
5. The checkboxes should be in the opposite state to step (3)
6. Back in settings, select "merged" or "all" mode
7. Both checkboxes in beam center finder tab should be enabled and checked

Fixes #25997 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py [hunk 1]
```diff
@@ -878,6 +882,17 @@ def on_save_other(self):
...
+        mode = get_reduction_mode_from_gui_selection(selection)
+        if mode == ISISReductionMode.HAB:
+            self._beam_centre_presenter.update_hab_selected()
+        elif mode == ISISReductionMode.LAB:
+            self._beam_centre_presenter.update_lab_selected()
+        else:
+            self._beam_centre_presenter.update_all_selected()
+
     # def _validate_rows(self):
     #     """
```
Comment (hsautessella) [inline scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py]: If this commented out method is unused, should it be deleted?

File: scripts/Interface/ui/sans_isis/beam_centre.py [hunk 2]
```diff
@@ -155,6 +155,18 @@ def set_run_button_to_normal(self):
         self.run_button.setText("Run")
         self.run_button.setEnabled(True)
 
+    def enable_update_hab(self, enabled):
+        if enabled:
+            self.update_hab_check_box.setEnabled(True)
+            self.update_hab_check_box.setChecked(True)
+        else:
+            self.update_hab_check_box.setChecked(False)
+            self.update_hab_check_box.setEnabled(False)
```
Comment (hsautessella) [inline scripts/Interface/ui/sans_isis/beam_centre.py]: If `enabled` is True or False can this method be condensed to just:
```python
def enable_update_hab(self, enabled):
    self.update_hab_check_box.setEnabled(enabled)
    self.update_hab_check_box.setChecked(enabled)
```

**Review by ghost — CHANGES_REQUESTED** [review]
Looks good and works as described, just 1 or 2 minor comments

File: scripts/Interface/ui/sans_isis/beam_centre.py (hunk 2 above)
Comment (TTitcombe) [reply scripts/Interface/ui/sans_isis/beam_centre.py]: Yep, this was left over from testing when I thought that the checkboxes could only be checked/unchecked when they were enabled. I'll change this

File: scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py (hunk 1 above)
Comment (TTitcombe) [reply scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py]: Git history shows it was commented out at least 11 months ago, so it's probably safe to remove

**Review by ghost — DISMISSED** [review]

**Comment by gemmaguest** [comment]
@TTitcombe Could you resolve the conflicts?

**Pull request merged**